runtime/src/main/java/org/corfudb/runtime/clients/NettyClientRouter.java

```diff
@@ -363,7 +388,12 @@ private void channelConnectionFutureHandler(@Nonnull ChannelFuture future,
             // a sleep period.
             if (!shutdown) {
                 log.info("connectAsync[{}]: Channel connection failed, reconnecting...", node);
-                Sleep.sleepUninterruptibly(parameters.getConnectionRetryRate());
+                if (sslAuthenticationFailure.get()) {
+                    retryPolicy.nextWait();
+                } else {
+                    retryPolicy.reset();
```

Comment:
Consider turning ExponentialBackoffRetry to an immutable class and assigning a new instance rather than reseting.